webxrlayers-1.bs

```diff
@@ -0,0 +1,730 @@
+  attribute float biasY;
+};
+</pre>
+
+The <dfn attribute for="XREquirectLayer">stereo</dfn> attribute returns if the layer is rendered in stereo. 
+
+The <dfn attribute for="XREquirectLayer">transform</dfn> attribute sets and returns the offset and orientation relative to the
+<dfn attribute for="XREquirectLayer">referenceSpace</dfn> attribute.
+
+The <dfn attribute for="XREquirectLayer">radius</dfn> attribute is the non-negative radius of the sphere. Values of less than or 
+equal to zero or infinity are treated as an infinite sphere.
+
+The <dfn attribute for="XREquirectLayer">scaleX</dfn> and <dfn attribute for="XREquirectLayer">scaleY</dfn> attributes 
+set and return a scale of the texture coordinates after the mapping to 2D.
+
+The <dfn attribute for="XREquirectLayer">biasX</dfn> and <dfn attribute for="XREquirectLayer">biasY</dfn> attributes 
+set and return a bias of the texture coordinates after the mapping to 2D.
+
+XRCubeLayer {#xcubelayertype}
+-----------
+A XREquirectLayer renders a layer where the [=XR Compositor=] renders directly from a cubemap.
+
+ISSUE: this section needs clarification
+
+<pre class="idl">
+interface XRCubeLayer : XRLayer {
+  readonly attribute boolean stereo;
+  attribute XRReferenceSpace referenceSpace;
+  attribute DOMPoint orientation;
+};
+</pre>
+
+The <dfn attribute for="XRCubeLayer">stereo</dfn> attribute returns if the layer is rendered in stereo. 
+
+The <dfn attribute for="XRCubeLayer">orientation</dfn> attribute sets and returns the orientation relative to the
+<dfn attribute for="XRCubeLayer">referenceSpace</dfn> attribute.
+
+Rendering {#rendering}
+=========
+
+
+XRSubImage {#xrsubimagetype}
+----------
+The {{XRSubImage}} object represents what viewport of the GPU texture to use.
+
+<pre class="idl">
+interface XRSubImage {
+  readonly attribute XRViewport viewport;
+};
+</pre>
+
+NOTE: this class is designed to accomodate future extensions
+
+The <dfn attribute for="XRSubImage">viewport</dfn> attribute returns the {{XRView}} to use when rendering the sub image.
+
+ISSUE: is imageIndex needed?
+
+XRWebGLSubImage {#xrwebglsubimagetype}
+---------------
+The {{XRWebGLSubImage}} object is used during rendering of the layer.
+
+<pre class="idl">
+interface XRWebGLSubImage : XRSubImage {
+  readonly attribute WebGLTexture colorTexture;
+  readonly attribute WebGLTexture? depthStencilTexture;
+  readonly attribute unsigned long? imageIndex;
+};
+</pre>
+
+The <dfn attribute for="XRWebGLSubImage">colorTexture</dfn> attribute returns the color {{WebGLTexture}} for the {{XRLayer}}.
+
+The <dfn attribute for="XRWebGLSubImage">depthStencilTexture</dfn> attribute returns the depth/stencil {{WebGLTexture}} for the {{XRLayer}}. 
+If the layer was created without depth/stencil, this attribute returns null.
+
+The <dfn attribute for="XRWebGLSubImage">imageIndex</dfn> attribute returns the offset into the texture array. Valid only for layers 
+that were requested with TEXTURE_2D_ARRAY.
```

Comment:
It's a bit of a pain to do, because the WebGL specs are not well formatted for it, but we should try to link WebGL enums and the like back to the specs where they are defined. See the WebXR spec's XRWebGLLayer section for examples.